### PR TITLE
Add json schema and validation for fields

### PR DESCRIPTION
**Pull request by raethlo**

### CODE REVIEW

File: src/panoramic/cli/schemas/field.schema.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    "field_type": {
+      "description": "Type of field",
+      "type": "string"
+    },
+    "description": {
+      "description": "Free text describing the field",
+      "type": "string"
+    },
+    "calculation": {
+      "description": "Formula describing how's the field calculated",
```
Comment (mrknmc) [inline src/panoramic/cli/schemas/field.schema.json:41]:
```suggestion
      "description": "Formula describing how the field is calculated",
```

File: src/panoramic/cli/schemas/field.schema.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      "description": "Display format to be used for the field in Pano UI",
+      "type": "string"
+    },
+    "aggregation": {
+      "description": "Type of aggregation to be used on the field",
+      "type": "object",
+      "properties": {
+        "type": {
+          "description": "Type of the aggregation",
+          "type": "string"
```
Comment (mrknmc) [inline src/panoramic/cli/schemas/field.schema.json:54]: No params? Can we make `type` required and follow the validation logic from here? https://github.com/panoramichq/diesel-service/blob/3fca007f77e1012c2c7cb70ae8f374484c120b11/core/model/data/taxonomy_aggregations.py#L77-L92

File: src/panoramic/cli/validate.py [hunk 3]
```diff
@@ -64,6 +72,33 @@ def _validate_file(fp: Path, schema: Dict[str, Any]):
...
 
 
+def _validate_fields(fields: Iterable[Tuple[Dict[str, Any], Path]]) -> List[ValidationError]:
+    field_paths_by_id: Dict[Tuple, List[Path]] = defaultdict(list)
+    errors = []
+    for field_data, field_path in fields:
+        try:
+            _validate_data(field_data, JsonSchemas.field())
+            field = PanoField.from_dict(field_data)
+            field_paths_by_id[field.id].append(field_path)
```
Comment (mrknmc) [inline src/panoramic/cli/validate.py:82]: It's kind of non obvious what's going on here with the company/dataset scoping... can we either change `id` to be `Tuple[Optional[str], str]` or even better, use `field_paths_by_id[(field.data_source, field.slug)] = ...` here?

That way, you can replace the iteration below with
```
for (data_source, field_slug), paths in field_paths_by_id.items():
```
?

File: src/panoramic/cli/errors.py [hunk 4]
```diff
@@ -232,6 +232,21 @@ def __init__(self, *, model_name: str, paths: List[Path]) -> None:
...
+    """Two local models use the same model name."""
+
+    def __init__(self, *, field_slug: str, dataset_slug: Optional[str], paths: List[Path]) -> None:
+        try:
+            paths = [path.relative_to(Path.cwd()) for path in paths]
+        except ValueError:
+            pass  # Use relative path when possible
+
+        path_lines = ''.join(f'\n  in {path}' for path in paths)
+        dataset_message = f' under dataset {dataset_slug} ' if dataset_slug else ''
```
Comment (mrknmc) [inline src/panoramic/cli/errors.py:246]:
```suggestion
        dataset_message = f' in dataset {dataset_slug} ' if dataset_slug else ''
```

**Review by mrknmc — CHANGES_REQUESTED** [review]
Very nice, I like the large amount of test cases!

File: src/panoramic/cli/schemas/field.schema.json (hunk 2 above)
Comment (raethlo) [reply src/panoramic/cli/schemas/field.schema.json:54]: 👍 I'll fix it in the base with standalone classes for the aggregation type and add the validation in.

**Review by mrknmc — APPROVED** [review]

File: src/panoramic/cli/validate.py (hunk 3 above)
Comment (raethlo) [reply src/panoramic/cli/validate.py]: Done c0ffb48.

**Pull request merged**